google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/SpannerInterceptorProvider.java

```diff
@@ -51,6 +52,7 @@ public static SpannerInterceptorProvider createDefault(OpenTelemetry openTelemet
     defaultInterceptorList.add(
         new LoggingInterceptor(Logger.getLogger(GapicSpannerRpc.class.getName()), Level.FINER));
     defaultInterceptorList.add(new HeaderInterceptor(new SpannerRpcMetrics(openTelemetry)));
+    defaultInterceptorList.add(new TraceContextInterceptor(openTelemetry));
```

Comment:
Is it possible to make the addition of this interceptor conditional on OpenTelemetry actually being used for this Spanner instance?

Reply:
Yes, it can be made conditional. We are also adding a option in SpannerOptions to enable server side tracing (https://github.com/googleapis/java-spanner/pull/3143). What we can do is if only user has opted-in by enabling the server side tracing then add this interceptor. Does it work?

cc: @dsimil

Reply:
@olavloite As discussed offline, made the use of TraceContextInterceptor conditional. This ClientInterceptor will only be used if client has opted-in for server side tracing explicitly by setting `enableServerSideTracing` option in SpannerOptions. 

Merged the changes for opt-in flag in this PR for using it to make use of interceptor conditional.